src/litdata/utilities/dataset_utilities.py

```diff
@@ -91,13 +93,66 @@ def _should_replace_path(path: Optional[str]) -> bool:
     return path.startswith("/teamspace/datasets/") or path.startswith("/teamspace/s3_connections/")
 
 
+def _read_updated_at(input_dir: Optional[Dir]) -> str:
+    """Read last updated timestamp from index.json file."""
+    last_updation_timestamp = ""
+    index_json_content = None
+    assert isinstance(input_dir, Dir)
+
+    if input_dir.path is not None and os.path.exists(os.path.join(input_dir.path, _INDEX_FILENAME)):
+        # read index.json file and read last_updation_timestamp
+        index_json_content = load_index_file(input_dir.path)
+    elif input_dir.url is not None:
+        assert input_dir.url is not None
+        # download index.json file and read last_updation_timestamp
+        with tempfile.TemporaryDirectory() as tmp_directory:
+            temp_index_filepath = os.path.join(tmp_directory, _INDEX_FILENAME)
+            downloader = get_downloader_cls(input_dir.url, tmp_directory, [])
+            downloader.download_file(os.path.join(input_dir.url, _INDEX_FILENAME), temp_index_filepath)
+
+            index_json_content = load_index_file(tmp_directory)
+
+    if index_json_content is not None and "updated_at" in index_json_content:
+        last_updation_timestamp = index_json_content["updated_at"]
+
+    return last_updation_timestamp
+
+
+def _clear_cache_dir_if_updated(input_dir_hash_filepath: str, updated_at_hash: str) -> None:
+    """Clear cache dir if it is updated.
+
+    If last_updated has changed and /cache/chunks/{HASH(input_dir.url)} isn't empty, we remove all the files and then
+    create the cache.
+
+    """
+    if os.path.exists(input_dir_hash_filepath):
+        # check if it only contains one directory with updated_at_hash
+        dir_list = os.listdir(input_dir_hash_filepath)
+        if not (len(dir_list) == 1 and dir_list[0] == updated_at_hash):
+            shutil.rmtree(input_dir_hash_filepath)
+
+
 def _try_create_cache_dir(input_dir: Optional[str]) -> Optional[str]:
-    hash_object = hashlib.md5((input_dir or "").encode())  # noqa: S324
+    resolved_input_dir = _resolve_dir(input_dir)
+    updated_at = _read_updated_at(resolved_input_dir)
+
+    if updated_at == "":
+        # for backward compatibility, use the input_dir for hashing (if no timestamp is found)
+        updated_at = input_dir if input_dir else ""
+
+    dir_url_hash = hashlib.md5((resolved_input_dir.url or "").encode()).hexdigest()  # noqa: S324
+    updated_at_hash = hashlib.md5((updated_at).encode()).hexdigest()  # noqa: S324
```

Comment:
Maybe, we don't even need to hash `last_updated`. This would make it more clear which one we are using.